release/models/flex-algo/openconfig-flexalgo.yang

```diff
@@ -0,0 +1,397 @@
+       If the required calculation type is Shortest Path First,
+       the value 0 SHOULD appear in this field.";
+  }
+
+  grouping algorithm-attributes {
+    description
+      "All of the possible attributes for an flexible algorithm
+       definition";
+    leaf name {
+      type string;
+      description
+        "Flexible Algorithm name";
+    }
+    leaf flex-algo-id {
+      type flex-algo-id;
+      mandatory true;
+      description
+        "Flex Algorithm Identifier";
+    }
+    leaf flex-algo-description {
+      type string;
+      description
+        "Description of this flex-algo";
+    }
+    leaf metric-type {
+      type flex-algo-metric-type;
+      default "IGP";
+      description
+        "Indicates which is the corresponding metric type for
+         this algorithm";
+    }
+    leaf calc-type {
+      type flex-algo-calc-type;
+      description
+        "Indicates the calculation type, if advertised";
+    }
+    leaf priority {
+      type uint8;
+      default "237";
+      description
+        "Indicates the priority for this algorithm";
+    }
+
+    leaf-list exclude {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups to exclude in path calculation.";
+    }
+    leaf-list include-all {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups, all of which must be included";
+    }
+    leaf-list include-any {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups, one of which must be included";
+    }
+    leaf-list srlg-names {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/srlgs/srlg/name";
+      }
+      description
+        "list of references to named SRLGs to be excluded";
+    }
+    leaf flags-tlv {
+      type boolean;
+      description
+        "Advertise flex-algo definition flags";
+    }
+  }
+
+  grouping flex-algo-definition-top {
+    description
+      "Top level grouping for flexible Algorithms and their definitions";
+    list algorithm {
+      key "name";
+      description
+        "List of Flexible Algorithms";
+      leaf name {
+        type leafref {
+          path "../config/name";
+        }
+        description
+          "Name for a Flexible Algorithm";
+      }
+      container config {
+        description
+          "Attributes for a Flexible Algorithm";
+        uses algorithm-attributes;
+      }
+      container state {
+        config false;
+        description
+          "Attributes for a Flexible Algorithm";
+        uses algorithm-attributes;
+      }
+    }
+  }
```

Comment:
I'm not sure what is this used for. There can be many FADs advertised per area from different sources, so you may many of these. Out of the many, one is selected as a winner and becomes the algo definition. Why do you want to store each one here?